staging/src/kubesphere.io/utils/helm/executor.go

```diff
@@ -36,41 +56,496 @@ type Executor interface {
+		args = append(args, "--kubeconfig", e.kubeConfigPath())
+	}
+
+	// Post render, add annotations or labels to resources
+	if len(e.labels) > 0 || len(e.annotations) > 0 {
+		args = append(args, "--post-renderer", filepath.Join(workspaceBase, postRenderExecFile))
+	}
+
+	if klog.V(8) {
+		// output debug info
+		args = append(args, "--debug")
+	}
+
+	name, err := e.createConfigMap(ctx, chartName, chartData, values)
+	if err != nil {
+		return "", err
+	}
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: e.namespace,
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32(1),
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "helm",
+							Image:           e.helmImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         []string{"helm"},
+							Args:            args,
+							WorkingDir:      workspaceBase,
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "source",
+									MountPath: workspaceBaseSource,
+								},
+								{
+									Name:      "data",
+									MountPath: workspaceBase,
+								},
+							},
+							Lifecycle: &corev1.Lifecycle{
+								PostStart: &corev1.LifecycleHandler{
+									Exec: &corev1.ExecAction{
+										Command: []string{"/bin/sh", "-c", fmt.Sprintf("cp -r %s/. %s", workspaceBaseSource, workspaceBase)},
+									},
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "source",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: name,
+									},
+									DefaultMode: pointer.Int32(0755),
+								},
+							},
+						},
+						{
+							Name: "data",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					RestartPolicy:                 corev1.RestartPolicyNever,
+					TerminationGracePeriodSeconds: new(int64),
+					ServiceAccountName:            fmt.Sprintf("helm-%s", e.namespace),
+				},
+			},
+		},
+	}
+
+	if err = e.Create(ctx, job); err != nil {
+		return "", err
+	}
+	return name, nil
 }
 
 // Uninstall uninstalls the specified chart, returns the name of the Job that executed the task.
 func (e *executor) Uninstall(ctx context.Context) (string, error) {
-	return "", nil
+	if _, err := e.status(); err != nil && err.Error() == statusNotFoundFormat {
+		// already uninstalled
+		return "", nil
+	}
+
+	args := []string{
+		"uninstall",
+		e.releaseName,
+		"--namespace",
+		e.namespace,
+	}
+	if e.dryRun {
+		args = append(args, "--dry-run")
+	}
+
+	name := generateName(e.releaseName)
+	if e.kubeConfigPath() != "" {
+		args = append(args, "--kubeconfig", e.kubeConfigPath())
+
+		configMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
```

Comment:
```suggestion
				GenerateName:      fmt.Sprintf("helm-executor-%s-",e.releaseName),
```

Reply:
We can't use this cause we need to return a definite name

Reply:
The name field will return after creation.

Reply:
For convenience, I set the Job's corresponding ConfigMap name to the same name as the Job itself, I don't think it's easier and more convenient to use `GenerateName`